packages/protvista-utils/src/withMargin.js

```diff
@@ -0,0 +1,48 @@
+const sides = ["left", "right", "top", "bottom"];
+const marginSides = sides.map((side) => `margin${side}`);
+
+const withMargin = (
+  Element,
+  options = {
+    initialValue: {
+      top: 0,
+      bottom: 0,
+      left: 0,
+      right: 0,
+    },
+  }
+) => {
+  for (const side of sides) {
+    Object.defineProperty(Element.prototype, `margin${side}`, {
```

Comment:
Maybe this should be applied on `ElementWithMargin` and not on `Element`, as it will modify the original element's prototype unnecessarily as the one that's really going to be used will be the returned `ElementWithMargin`?

Reply:
Cool, I see there is a potential for side effects with the current code, I'll change it as you suggest.